### PR TITLE
Improve correctness of UnusedPrivateProperty

### PR DESCRIPTION
**Pull request by luanpotter**
Fixes https://github.com/detekt/detekt/issues/5934

Basically ignore package declarations when counting references for the purposes of identifying Unused Private Properties.

## Followups

I believe there are many other edge cases that could be considered; for example, the right-hand side of a dot-expression is also being considered a use reference:

```kotlin
class A (val a: String = "foo")

fun main() {
    val a = 1
    println(A().a)
}
```

Should cause an error, but it doesn't. I'm not sure if there is a better solution here other than exploring each edge case individually, but I am happy to chip down on at least the more obvious ones on followup PRs.

### CODE REVIEW

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivateProperty.kt [hunk 1]
```diff
@@ -131,7 +126,20 @@ private class UnusedPrivatePropertyVisitor(private val allowedNames: Regex) : De
...
     override fun visitReferenceExpression(expression: KtReferenceExpression) {
-        nameAccesses.add(expression.text.removeSurrounding("`"))
+        if (!skipNode(expression)) {
+            nameAccesses.add(expression.text.removeSurrounding("`"))
+        }
         super.visitReferenceExpression(expression)
     }
+
+    private fun skipNode(expression: KtReferenceExpression): Boolean {
+        return when {
```
Comment (luanpotter) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivateProperty.kt:136]: note: adding a bit of structure as we can followup with more exceptions and rules for other edge cases

Comment (luanpotter) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivateProperty.kt:143]: example of a followup: ignore import statements as well

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivatePropertySpec.kt [hunk 2]
```diff
@@ -808,4 +808,41 @@ class UnusedPrivatePropertySpec(val env: KotlinCoreEnvironment) {
...
+                fun main() {
+                    val unused = 1
+                    println("foo")
+                }
+            """.trimIndent()
+            assertThat(subject.lint(code)).hasSize(1)
+        }
+
+        @Test
+        fun `package declarations are ignored`() {
```
Comment (schalkms) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivatePropertySpec.kt:837]: I do think we only need this test to be added. The other cases are covered by the existing tests.

**Review by schalkms — COMMENTED** [review]
Thanks for the quick fix!
Besides the redundant two tests, this PR looks very good!

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivatePropertySpec.kt (hunk 2 above)
Comment (luanpotter) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivatePropertySpec.kt:815]: good point! I had skimmed the file for these tests but initially failed to see them nested in the inner classes.
I removed the other two.

on that note though, lmk of including my 1 test under a new "irrelevant references are ignored" nested class (it seems all tests are nested somewhere)
I was planning on following this up with some other edge cases (eg import statements should be ignored as well)
but I am happy to put the test nested somewhere else if that makes sense, just lmk where it better fits

Comment (schalkms) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivatePropertySpec.kt:815]: Regarding test under new nested class 👍
Regarding follow up --> please submit a new PR, it's easier to review 👍

**Review by schalkms — APPROVED** [review]

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivatePropertySpec.kt (hunk 2 above)
Comment (luanpotter) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivatePropertySpec.kt:815]: yes, def meant followup PRs! thanks for reviewing so swiftly :)

**Review by BraisGabin — APPROVED** [review]
